src/mono/mono/mini/method-to-ir.c

```diff
@@ -8899,6 +8899,12 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 			ins->sreg2 = sp [1]->dreg;
 			type_from_op (cfg, ins, sp [0], sp [1]);
 			CHECK_TYPE (ins);
+#ifdef TARGET_RISCV
+			// Don't multiply r4 with r8 in RISC-V
```

Comment:
Why is this riscv only ?

Reply:
actually, it doesn't means it is riscv only. I am not that familiar with other Arch. But in RISC-V Arch, There are some cases may lead to r4 multiply by r8 (sry I forget the specific case). So I add a code to process widen type. Considering defensive programming, I didn't want to cause errors in other architectures, so I used macro definitions.

Do you think I should make it as a general processing?

Reply:
Yes, it could be enabled on all platform.